KEYCLOAK-16923 Revising Authorization Services Guide for Red Hat Standards

andymunro opened the pull request:
@pdrozd This is the last guide in the documentation, which is now revised to meet Red Hat standards for findability and reusability of topics.

andymunro committed:
Removing include for the conscious language note

nehachopra27 commented on authorization_services/topics/hello-world-create-realm.adoc:
```diff
@@ -1,21 +1,26 @@
+. Click *Save*.
 +
 .Add User
 image:{project_images}/getting-started/hello-world/create-user.png[alt="Add User"]
```
@andymunro : The casing of image heading is all caps throughout this guide. Does it needs to match the other headings? Please check and if required then it needs to be changed throughout.

andymunro replied:
Thanks.  You are right, @nehachopra27.  I will change it to sentence case, meaning lower case except for the first word or a proper name.

nehachopra27 commented on authorization_services/topics/policy-evaluation-api.adoc:
```diff
@@ -54,7 +54,7 @@ Policies determine this by invoking the `grant()` or `deny()` methods on an `Eva
 
 For more information about the Evaluation API see the {apidocs_link}[JavaDocs].
```
@andymunro: Should this be changed to Additional Resources?

nehachopra27 commented on authorization_services/topics/permission-create-resource.adoc:
```diff
@@ -1,9 +1,9 @@
+To create a new resource-based permission, select *Resource-based* in the item list in the upper right corner of the permission listing.
 
 .Add Resource-Based Permission
 image:{project_images}/permission/create-resource.png[alt="Add Resource-Based Permission"]
```
Image casing

andymunro replied:
I think this one should remain initial capitals to match the UI.

nehachopra27 commented on authorization_services/topics/permission-create-scope.adoc:
```diff
@@ -1,9 +1,9 @@
+To create a new scope-based permission, select *Scope-based* in the item list in the upper right corner of the permission listing.
 
 .Add Scope-Based Permission
 image:{project_images}/permission/create-scope.png[alt="Add Scope-Based Permission"]
```
image title casing

andymunro replied:
I think this one should remain initial capitals to match the UI.

nehachopra27 commented on authorization_services/topics/policy-aggregated-policy.adoc:
```diff
@@ -1,9 +1,9 @@
+To create a new aggregated policy, select *Aggregated* in the item list located in the right upper corner of the policy listing.
 
 .Add an Aggregated Policy
 image:{project_images}/policy/create-aggregated.png[alt="Add Aggregated Policy"]
```
image title casing

andymunro replied:
I think this one should remain initial capitals to match the UI.

nehachopra27 commented on authorization_services/topics/policy-client-policy.adoc:
```diff
@@ -1,9 +1,9 @@
+To create a new client-based policy, select *Client* in the item list in the upper right corner of the policy listing.
 
 .Add a Client-Based Policy
 image:images/policy/create-client.png[alt="Add Client-Based Policy"]
```
image title casing

andymunro replied:
I think this one should remain initial capitals to match the UI.

nehachopra27 commented on authorization_services/topics/policy-client-scope-policy.adoc:
```diff
@@ -1,11 +1,11 @@
+To create a new client scope-based policy, select *Client Scope* in the item list in the upper right corner of the policy listing.
 
 .Add Client Scope-Based Policy
 image:images/policy/create-client-scope.png[alt="Add Client Scope-Based Policy"]
```
image title casing

andymunro replied:
I think this one should remain initial capitals to match the UI.

nehachopra27 commented on authorization_services/topics/policy-role-policy.adoc:
```diff
@@ -1,13 +1,13 @@
+To create a new role-based policy, select *Role* in the item list in the upper right corner of the policy listing.
 
 .Add Role-Based Policy
 image:{project_images}/policy/create-role.png[alt="Add Role-Based Policy"]
```
image title casing

nehachopra27 commented on authorization_services/topics/hello-world-create-resource-server.adoc:
```diff
@@ -1,29 +1,40 @@
+
+. Fill in the *Client ID*, *Client Protocol*, and *Root URL* fields.
 +
 .Create Client Application
```
image title casing here is not read from alt text, it is reading the . title.

nehachopra27 commented on authorization_services/topics/policy-client-scope-policy-required-client-scope.adoc:
```diff
@@ -8,4 +8,4 @@ image:images/policy/create-client-scope.png[alt="Example of Required Client Scop
 
```
Image title casing

nehachopra27 commented on authorization_services/topics/policy-group-policy.adoc:
```diff
@@ -1,12 +1,12 @@
 
-.Add Group-Based Policy
-image:{project_images}/policy/create-group.png[alt="Add Group-Based Policy"]
+Group-Based Policy
```
Image title casing

andymunro replied:
I think this one should remain initial capitals to match the UI.

nehachopra27 commented on authorization_services/topics/policy-user-policy.adoc:
```diff
@@ -1,9 +1,9 @@
+To create a new user-based policy, select *User* in the item list in the upper right corner of the policy listing.
 
 .Add a User-Based Policy
 image:{project_images}/policy/create-user.png[alt="Add User-Based Policy"]
```
Image title casing

andymunro replied:
I think this one should remain initial capitals to match the UI.

nehachopra27 commented on server_admin/topics/realms/keys.adoc:
```diff
@@ -17,6 +17,10 @@ When a realm is created a key pair and a self-signed certificate is automaticall
 . Click *Passive* to view passive keys.
 . Click *Disabled* to view disabled keys.
 
+<<<<<<< HEAD
```
Conflict has been pushed here.

nehachopra27 commented on server_admin/topics/realms/keys.adoc:
```diff
@@ -37,6 +41,7 @@ Rotating keys also applies to offline tokens. To make sure they are updated, the
 
 ==== Adding a generated keypair
 
+<<<<<<< HEAD
```
PLease check the conflict here

andymunro commented:
@nehachopra27 Can you check this again.  There are still build errors about a missing "context" variable. I will ask development about these errors.  However, I addressed your feedback. Thanks.

andymunro commented:
@nehachopra27  Any more input on this guide?

andymunro commented:
@nehachopra27 I improved the use of sentence case which allows for initial capital letters to match the names on the screen.  It is hard to be consistent completely. This is probably the best I can do at this release.

nehachopra27 commented on authorization_services/topics/policy-role-policy.adoc:
```diff
@@ -33,4 +33,8 @@ Specifies which *client* roles are permitted by this policy. To enable this fiel
+
+[role="_additional-resources"]
+.Additional resources
+* <<_policy_logic, Postive and negative logic>>
```
@andymunro: Positive is misspelled here.

nehachopra27 commented on authorization_services/topics/policy-time-policy.adoc:
```diff
@@ -50,6 +50,10 @@ Defines the minute that access must be granted. You can also specify a range of
+
+[role="_additional-resources"]
+.Additional resources
+* <<_policy_logic, Postive and negative logic>>
```
Positive is misspelled here.

nehachopra27 commented on authorization_services/topics/policy-user-policy.adoc:
```diff
@@ -25,4 +25,8 @@ Specifies which users are given access by this policy.
+
+[role="_additional-resources"]
+.Additional resources
+* <<_policy_logic, Postive and negative logic>>
```
Positive is misspelled here.

nehachopra27 replied:
This is misspelled in all the subsection policies.

nehachopra27 commented:
@andymunro : I reviewed the changes. I had one additional comment which I raised, rest everything looks good.

andymunro commented:
@nehachopra27 I made that correction. Thanks.  I think the errors reported are invalid.  The link identified as broken does take you to the Instagram topic in the Server Administration.

nehachopra27 commented:
@pdrozd : This guide is verified.

andymunro commented:
@mposolda Would you be able to review this guide, which is approved by QE?

abstractj reviewed (approved):
LGTM

abstractj reviewed (changes requested):
@andymunro that was too fast. Could you please address the failures below?

```
Error:  Failures: 
Error:    GuideTest.checkVariables:18->checkFailures:51 Variables not found  (1):
		- context
```

//cc @nehachopra27

andymunro commented:
Thanks, @abstractj.  Yes, I have been trying to locate the cause of this failure.  The log file shows all the books built with no errors.

abstractj commented:
@andymunro I noticed that our [admin server guide](https://raw.githubusercontent.com/keycloak/keycloak-documentation/21d73659caa9b2356f15bd253a9078a922cfb159/server_admin/master.adoc) uses a similar approach, the only difference is:

### Admin server guide

```:context: server_administration_guide``` - No colon

### Changes here

```:context: authorization_services_guide:``` - With colon at the end

I'm no expert on this, but maybe that's the issue.

andymunro commented:
Thanks, @abstractj.  I tried removing the colon and then even removing the "context" code, however it still generates this error.  The product version does not have this error and the only difference in the books is the master.adoc versus index.adoc.  Maybe @pskopek has an idea.

andymunro commented:
This PR is replace by this one: https://github.com/keycloak/keycloak-documentation/pull/1358
